src/scripting/exposed_object.hpp

```diff
@@ -0,0 +1,70 @@
+//  SuperTux -- ExposedObject class
+//  Copyright (C) 2016 Tobias Markus <tobbi.bugs@googlemail.com>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef HEADER_SUPERTUX_SCRIPTING_EXPOSED_OBJECT_HPP
+#define HEADER_SUPERTUX_SCRIPTING_EXPOSED_OBJECT_HPP
+
+#include "scripting/scripting.hpp"
+#include "scripting/squirrel_util.hpp"
+#include "supertux/script_interface.hpp"
+#include "util/log.hpp"
+
+/**
+ * @class ExposedObject
+ * @param class S: GameObject class (e.g. Gradient)
+ * @param class T: Scripting class (e.g. scripting::Gradient)
+ */
+template<class S, class T>
+class ExposedObject : public ScriptInterface
+{
+private:
+  S* m_parent;
+
+public:
+  ExposedObject(S* parent) :
+    m_parent(parent)
+  {
+  }
+
+  void expose(HSQUIRRELVM vm, SQInteger table_idx)
+  {
+    auto name = this->m_parent->get_name();
+    if (name.empty())
+      return;
+
+    log_debug << "Exposing object " << name << std::endl;
+
+    auto object = new T(m_parent);
```

Comment:
According to [CODINGSTYLE](https://github.com/SuperTux/supertux/blob/master/CODINGSTYLE#L17), you want to use `std::unique_ptr<>` here.